### ISSUE TITLE
Starting Atom: Uncaught RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by icimasa** [atom/atom#11520]
I get this error everytime I start Atom if there is a tab open in the list that Atom wants to reopen, if there is not a tab open when starting Atom, everything is fine.. 

This issue has been since v. 1.7.0

I run Windows 10

C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\underscore-plus\node_modules\underscore\underscore.js:219

RangeError: Maximum call stack size exceeded
    at Array.indexOf (native)
    at Object._.contains._.include (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\underscore-plus\node_modules\underscore\underscore.js:219:68)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:275:14)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)
    at GrammarRegistry.module.exports.GrammarRegistry.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar-registry.js:180:23)
    at Grammar.module.exports.Grammar.grammarUpdated (C:\Users\masa\AppData\Local\atom\app-1.7.1\resources\app.asar\node_modules\first-mate\lib\grammar.js:279:21)

**Comment by infernocloud** [comment]
This same error is affecting me as well after the 1.7.2 update.

**Comment by thecolorblue** [comment]
I got this error in windows 7 (ultimate), with atom 1.7.2.

**Comment by jasongutt** [comment]
I got this error after the editor crashed and I restarted.

**Comment by orcpac7** [comment]
Got it in Atom 1.6.2 on Windows 7, SP1.
Note that I had edited a new file, saved it, exited Atom, found the file **empty** so restarted Atom, which opened the same file, I pasted code back in and attempted to save. 
It was at this point that I got the error.

**Comment by orcpac7** [comment]
OK, looked in Help and saw that it was downloading, clicked install update in lower-right of screen (status bar?) and it updated to 1.7.2. I'll keep an eye and see if I see this issue again...

**Comment by icimasa** [comment]
This error is still present in 1.7.2!!

**Comment by lee-dohm** [comment]
Everyone please try the following:

1. **Save your work**
1. Completely exit Atom
1. Launch Atom again using `atom --clear-window-state`

**Comment by icimasa** [comment]
Well that works, because then it clears open tabs, but therein lies the problem for me at least, if you have tabs open when starting atom you get the stack error!

**Comment by lee-dohm** [comment]
What kind of file are you opening @icimasa?

**Comment by icimasa** [comment]
@lee-dohm I open .cfm files

**Comment by lee-dohm** [comment]
@icimasa What if there is a single generic text file in the list of files Atom wants to reopen? Does the exception happen then?

**Comment by icimasa** [comment]
@lee-dohm I don't know if the error happend all the time when reopen, but I tried to just have som css files open and did som quick tests without being able to reproduce the error. I will do some more tests!

**Comment by icimasa** [comment]
@lee-dohm hi, I did some more testing this morning, I have 2 tabs open with .cfm files and the CFML(HTML) language pack, I switched both files to Plain Text, closed the editor and reopen without error, so I switched the open files back to CFML(HTML) and closed the editor and reopened and boom I got the error again. So it seems to be related to the CFML(HTML) language! Do I need to report this somewhere else?
//Martin
